onnxruntime/core/optimizer/initializer.cc

```diff
@@ -11,49 +10,313 @@
 #include "core/framework/tensor_external_data_info.h"
 #include "core/platform/env.h"
 
+#include <functional>
+
 namespace onnxruntime {
-Status Initializer::ReadExternalRawData(
-    const ONNX_NAMESPACE::TensorProto& tensor_proto, const Path& model_path, std::vector<char>& raw_data) {
-  ORT_RETURN_IF_NOT(
-      tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED &&
-          tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_STRING,
-      "External data type must not be UNDEFINED or STRING.");
 
-  ORT_RETURN_IF(
-      model_path.IsEmpty(),
-      "model_path must not be empty. Ensure that a path is provided when the model is created or loaded.");
+Initializer::Initializer(ONNX_NAMESPACE::TensorProto_DataType data_type,
```

Comment:
Are so many changes in this file related to this PR? If not, can they be separated to avoid noise?